TABTestKit/Classes/Apps/App.swift

```diff
@@ -14,11 +14,36 @@ import XCTest
 /// Using this instead of XCUIApplication() means you get the
 /// enhanced waits for the correct states when launching/backgrounding etc, inherited from BaseApp.
 public final class App: BaseApp {
+
+    public struct ScreenshotOption: OptionSet {
+        public let rawValue: Int
+
+        public init(rawValue: Int) {
+            self.rawValue = rawValue
+        }
+
+        public static let onFailure = ScreenshotOption(rawValue: 1 << 0)
+        public static let beforeStep = ScreenshotOption(rawValue: 1 << 1)
+        public static let afterStep = ScreenshotOption(rawValue: 1 << 2)
+        public static let none: ScreenshotOption = []
+        public static let all: ScreenshotOption = [.onFailure, .beforeStep, .afterStep]
+    }
 	
 	/// The shared App instance. Using a shared instance ensure consistency with launch arguments.
 	public static let shared = App()
+
+    public private(set) var screenshotOption: ScreenshotOption = .none
```

Comment:
I don't think this belongs in the app object/element, because it's a test option, so probably belongs in TABTestCase if anywhere

Reply:
processed feedback